rhel.Dockerfile

```diff
@@ -0,0 +1,129 @@
+    if [[ -f /tmp/root-local.tgz ]]; then mkdir -p /root/.local && tar xf /tmp/root-local.tgz -C /root/.local/ && rm -fr /tmp/root-local.tgz; fi && \
+    /usr/bin/pip3.6 install --user yq jsonschema && \
+    # could be installed in /opt/app-root/src/.local/bin or /root/.local/bin
+    for d in /opt/app-root/src/.local /root/.local; do \
+      if [[ -d ${d} ]]; then \
+        cp ${d}/bin/yq ${d}/bin/jsonschema /usr/local/bin/; \
+        pushd ${d}/lib/python3.6/site-packages/ >/dev/null; \
+          cp -r PyYAML* xmltodict* yaml* yq* jsonschema* /usr/lib/python3.6/site-packages/; \
+        popd >/dev/null; \
+      fi; \
+    done && \
+    chmod +x /usr/local/bin/yq && \
+    ln -s /usr/bin/python3.6 /usr/bin/python
+
+# for debugging only
+# RUN microdnf install -y util-linux && whereis python pip jq yq && python --version && jq --version && yq --version
+
+################# 
+# PHASE TWO: configure registry image
+#################
+
+COPY ./scripts/*.sh ./scripts/meta.yaml.schema /build/
+COPY /v3 /build/v3
+WORKDIR /build/
+
+# if only including the /latest/ plugins, apply this line to remove them from builder
+# RUN rm -fr $(find /build/v3 -name 'meta.yaml' | grep -v "/latest/")
+
+RUN ./check_plugins_location.sh v3 && \
+    ./set_plugin_dates.sh v3 && \
+    ./check_plugins_viewer_mandatory_fields.sh v3 && \
+    ./ensure_latest_exists.sh && \
+    ./index.sh v3 > /build/v3/plugins/index.json && \
+    chmod -c -R g+rwX /build
+
+################# 
+# PHASE THREE: create ubi8-minimal image with httpd
+################# 
+
+# Build registry, copying meta.yamls and index.json from builder
+# https://access.redhat.com/containers/?tab=tags#/registry.access.redhat.com/rhscl/httpd-24-rhel7
+FROM registry.access.redhat.com/rhscl/httpd-24-rhel7:2.4-104 AS registry
+USER 0
+
+# BEGIN these steps might not be required
+RUN sed -i /etc/httpd/conf/httpd.conf \
+    -e "s,Listen 80,Listen 8080," \
+    -e "s,logs/error_log,/dev/stderr," \
+    -e "s,logs/access_log,/dev/stdout," \
+    -e "s,AllowOverride None,AllowOverride All," && \
+    chmod a+rwX /etc/httpd/conf /run/httpd
+EXPOSE 80
+STOPSIGNAL SIGWINCH
+ENTRYPOINT ["/usr/sbin/httpd", "-D", "FOREGROUND"]
+# END these steps might not be required
+
+COPY README.md .htaccess /var/www/html/
+COPY --from=builder /build/v3 /var/www/html/v3
+
+# Offline build: cache .theia and .vsix files in registry itself and update metas
+FROM builder AS offline-builder
+
+# To only cache files from /latest/ folders, use ./cache_artifacts.sh v3 --latest-only 
+# and uncomment line above to remove files so they're not included in index.json -- RUN rm -fr $(find /build/v3 -name 'meta.yaml' | grep -v "/latest/")
+RUN ./cache_artifacts.sh v3 && chmod -R g+rwX /build
+
+FROM registry AS offline-registry
+USER 0
+
+# Offline registry: copy updated meta.yamls and cached extensions
+ENV SUMMARY="Red Hat CodeReady Workspaces plugin registry container" \
```

Comment:
it's not codeready workspaces there.

Reply:
can you just ignore it? it's not like you're using the rhel.Dockerfile to build che-plugin-registry, since you're using the alpine one :)

Reply:
No, I can't ;-)

Reply:
So I need these labels and envs in a separate file downstream, or applied via a transform/merge process. Fun. OK.